### PR TITLE
Fix a bug of flatten in ONNX to Relay converter

### PR DESCRIPTION
**Pull request by Oldpan**
Batch_flatten, as it's name suggests, should flatten each data samples of a batch.
So we don't need to enhance batch_flatten to support another axis except batch_size axis.
As a result, we can do an optional conversion (convert to batch_flatten when the flatten op matches, use reshape otherwise).

### CODE REVIEW

**Comment by Oldpan** [comment]
Related pull request #2843

**Comment by jroesch** [comment]
@zhreshold LGTM, thoughts?

File: python/tvm/relay/frontend/onnx.py [hunk 1]
```diff
@@ -335,6 +335,23 @@ class Reciprocal(OnnxOpConverter):
...
+    """ Operator converter for Flatten.
+    """
+
+    @classmethod
+    def _impl_v1(cls, inputs, attr, params):
+        axis = attr.get('axis', 1)
+        if axis == 1:
+            out = _op.nn.batch_flatten(inputs[0])
+        else:
+            newshape = [0]*(axis+1)
```
Comment (tqchen) [inline python/tvm/relay/frontend/onnx.py]: nit: space between operands ```[0] * (axis + 1)```

Comment (Oldpan) [reply python/tvm/relay/frontend/onnx.py]: OK

**Pull request merged**